source/common/network/connection_impl.cc

```diff
@@ -39,68 +40,88 @@ void ConnectionImplUtility::updateBufferStats(uint64_t delta, uint64_t new_total
   }
 }
 
-void DynamicSocketOptionsImpl::setSocketRecvBufferSize(uint32_t buffer_size) const {
-  int rc = 0;
-  if (io_handle_.fd() == -1) {
+Api::SysCallIntResult DynamicSocketOptionsImpl::setSocketRecvBufferSize(int buffer_size) const {
+  int rc = -1;
+  int errsrv = -1;
+  socklen_t option_len = sizeof(int);
+  if (-1 == io_handle_.fd()) {
     ENVOY_CONN_LOG(trace, "Stale socket file handle, operation Set Receive Buffer aborted", *this);
-    return;
+    return {rc, errsrv};
```

Comment:
How about EBADF for this one?  We want the error numbers to be informative.
Alternately will the syscall handle this for us?  If you're trying for a perf optimization to avoid the system call on an invalid FD let's move it to the system call, so it applies to all callers of setsockopt :-)

Reply:
Got it. I will move this to setsockopt call. It is my intention not to perform the system call, when we are already aware about the state of the file handle. However this will be handled by the system call. Please let me know if you do not want me to make this change and let the system call handle it.